codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -1165,4 +1132,36 @@ private boolean allowModifications() {
             return pendingIterations == 0;
         }
     }
+
+    /**
+     * Implementation of {@link PropertyKey} that specifies the index position of the property.
+     */
+    class DefaultPropertyKey implements PropertyKey {
+        private final int index;
+
+        DefaultPropertyKey(int index) {
+            this.index = index;
+        }
+    }
+
+    /**
+     * A registry of all stream property keys known by this connection.
+     */
+    private class PropertyKeyRegistry {
+        final List<DefaultPropertyKey> keys = new ArrayList<DefaultPropertyKey>(4);
```

Comment:
@Scottmitch @nmittler 

Regarding sharing some property keys among all connections... let's talk facts. A `DefaultPropertyKey` object takes 16 byte. We always have two for `FlowState` and say an application also uses two. So that's 64 byte per connection. If you do 10k connections per second that's 320KB in additional allocations per second, or 640KB if we would also do the static properties for users. In my opinion that's not something to worry about at this point. In GRPC for NOOP RPCs at ~10k streams per second we are allocating 300 - 400MB per second, with 4 connections. If you go 10000 connections your allocation rate will be a lot higher due to pipeline, handlers and so on. So we are looking at a optimization potential of far less than 1%, likely less than 0.1%. Not something to worry about.